File: cpp/0162.0_Find_Peak_Element.cpp
/*
执行用时：8 ms, 在所有 C++ 提交中击败了11.93% 的用户
内存消耗：8.7 MB, 在所有 C++ 提交中击败了8.48% 的用户
通过测试用例：63 / 63
*/

class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        return max_element(nums.begin(), nums.end()) - nums.begin();
    }
};

File: go/0162.0_Find_Peak_Element.go
/*
T: O(N)
S: O(1)

执行用时：0 ms, 在所有 Go 提交中击败了100.00% 的用户
内存消耗：2.7 MB, 在所有 Go 提交中击败了61.26% 的用户
通过测试用例：63 / 63
*/
func findPeakElement(nums []int) (idx int) {
    for i, v := range nums {
        if v > nums[idx] {
            idx = i
        }
    }
    return 
}
